### PR TITLE
test(mitm-addon): assert url rewrite tracking outside forward mock

### LINKED ISSUE #15494: test(mitm-addon): Assertions inside mock side_effect functions are swallowed by production except Exception handler
## Problem

Both forward_request (line 332) and fail_forward_request (line 427) mock side_effect functions contain assert and assert_pending calls that execute during request() processing, inside the try/except Exception block at auth.py:543 (within _apply_url_rewrite). If any of these assertions fail, the resulting AssertionError is caught by the broad except Exception, which silently treats it as a URL rewrite forward failure, sets a 502 firewall_error response, and returns False. The test then fails at later assertions (e.g., line 360 expects flow.metadata['auth_url_rewrite'] is True but the key is absent because the rewrite path was short-circuited). This produces confusing error messages that mask the real assertion failure. Fix: Remove the intermediate-state assertions from inside the mock side_effect functions and rely on the post-request() assertions already present in the test body (lines 359-381 and 454-466), which check the same state. If intermediate-state verification is required, capture state into a mutable container (list) inside the mock and assert on it after the with block.

**File:** /tmp/vm0-mitm-addon-review/crates/runner/mitm-addon/tests/test_request_handler_usage_tracking.py
**Lines:** 332-342, 427-437
**Severity:** medium
**Type:** test-defect

## Scenario

This was found during the daily mitm-addon audit on 2026-05-30.

## Suggested Fix

See description above for suggested approach.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by seven332**
## Summary
- Add a test-local async forward probe so URL rewrite tests can pause during forwarding and assert pending usage state outside the production exception boundary.
- Replace `AsyncMock(side_effect=...)` forward assertions with direct assertions before releasing the mocked forward call.
- Keep the success and forward-failure drain assertions after request completion.

Closes #15494

## Tests
- `PATH=/tmp/vm7-mitm-addon-deps/bin:$PATH PYTHONPATH=src:/tmp/vm7-mitm-addon-deps python3 -m pytest tests/test_request_handler_usage_tracking.py -k 'auth_url_rewrite'`\n- `PATH=/tmp/vm7-mitm-addon-deps/bin:$PATH PYTHONPATH=src:/tmp/vm7-mitm-addon-deps python3 -m pytest tests/test_request_handler_usage_tracking.py`\n- `PATH=/tmp/vm7-mitm-addon-deps/bin:$PATH PYTHONPATH=src:/tmp/vm7-mitm-addon-deps python3 -m ruff check tests/test_request_handler_usage_tracking.py`\n- `PATH=/tmp/vm7-mitm-addon-deps/bin:$PATH PYTHONPATH=src:/tmp/vm7-mitm-addon-deps python3 -m ruff format --check tests/test_request_handler_usage_tracking.py`\n- `PATH=/tmp/vm7-mitm-addon-deps/bin:$PATH PYTHONPATH=src:/tmp/vm7-mitm-addon-deps basedpyright -p .`

### CODE REVIEW

File: crates/runner/mitm-addon/tests/test_request_handler_usage_tracking.py [hunk 1]
```diff
@@ -447,10 +511,28 @@ async def fail_forward_request(*_args):
...
+            assert_pending(
+                usage_pending_path,
+                flows=1,
+                buffered=0,
+                reports=0,
+                flush_request_id="request-1",
+            )
+
+            probe.release.set()
+            await request_task
```
[the remark this answers is not in the document]
Comment (seven332) [reply crates/runner/mitm-addon/tests/test_request_handler_usage_tracking.py]: Keeping the normal-path await intentionally. The cleanup now only runs when the task is still pending, so this is not a duplicate await. The explicit await preserves propagation of unexpected request() failures after the in-flight assertions pass; relying only on cleanup would suppress those errors.

File: crates/runner/mitm-addon/tests/test_request_handler_usage_tracking.py [hunk 2]
```diff
@@ -352,10 +409,28 @@ async def forward_request(*_args):
...
```
[the remark this answers is not in the document]
Comment (seven332) [reply crates/runner/mitm-addon/tests/test_request_handler_usage_tracking.py]: Keeping the normal-path await intentionally. The cleanup now only runs when the task is still pending, so this is not a duplicate await. The explicit await preserves propagation of unexpected request() failures after the in-flight assertions pass; relying only on cleanup would suppress those errors.
